include/engine/routing_algorithms/many_to_many.hpp

```diff
@@ -143,10 +143,30 @@ class ManyToManyRouting final
                     (*result_table)[source_id * number_of_targets + target_id];
                 // check if new distance is better
                 const EdgeWeight new_distance = source_distance + target_distance;
-                if (new_distance >= 0 && new_distance < current_distance)
+                if (new_distance < 0)
                 {
-                    (*result_table)[source_id * number_of_targets + target_id] =
-                        (source_distance + target_distance);
+                    for (auto edge : super::facade->GetAdjacentEdgeRange(node))
```

Comment:
I feel this should be a utility function in `routing_base.hpp`. Something like:

`UpdateLoopWeight(NodeId node, EdgeWeight& loop_weight)`

Also can't we guarantee that each node has at most one relevant loop? In that case we can optimize this.